src/main/java/seedu/address/logic/parser/StatsCommandParser.java

```diff
@@ -30,11 +32,12 @@ public StatsCommand parse(String args) throws ParseException {
 
         try {
             ArgumentMultimap argMultimap =
-                    ArgumentTokenizer.tokenize(args, PREFIX_NUMBER_OF_DAYS_OR_MONTHS, PREFIX_MODE);
-            int numberOfDaysOrMonths = Integer.parseInt(argMultimap.getValue(PREFIX_NUMBER_OF_DAYS_OR_MONTHS).get());
+                    ArgumentTokenizer.tokenize(args, PREFIX_PERIOD_AMOUNT, PREFIX_PERIOD, PREFIX_MODE);
+            int periodAmount = Integer.parseInt(argMultimap.getValue(PREFIX_PERIOD_AMOUNT).get());
             String mode = argMultimap.getValue(PREFIX_MODE).get();
+            String period = argMultimap.getValue(PREFIX_PERIOD).get();
 
-            return new StatsCommand(numberOfDaysOrMonths, mode);
+            return new StatsCommand(periodAmount, period, mode);
         } catch (Exception e) {
```

Comment:
Catch a specific `Exception`. Catching all exceptions will cause potential bugs to not surface